presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -395,10 +395,19 @@ public long getLastSchedulingCycleRuntimeDelayMs()
         return currentTimeMillis() - lastSchedulingCycleRunTimeMs.get();
     }
 
-    private static int getQueriesQueuedOnInternal(InternalResourceGroup resourceGroup)
+    private int getQueriesQueuedOnInternal(InternalResourceGroup resourceGroup)
     {
         if (resourceGroup.subGroups().isEmpty()) {
-            return Math.max(Math.min(resourceGroup.getQueuedQueries(), resourceGroup.getSoftConcurrencyLimit() - resourceGroup.getRunningQueries()), 0);
+            int queuedQueries = resourceGroup.getQueuedQueries();
+            int runningQueries = resourceGroup.getRunningQueries();
+            if (isResourceManagerEnabled) {
+                ResourceGroupRuntimeInfo resourceGroupRuntimeInfo = resourceGroupRuntimeInfos.get().get(resourceGroup.getId());
+                if (resourceGroupRuntimeInfo != null) {
```

Comment:
Is this an error scenario to have  resourceGroupRuntimeInfo null for a resource group here? Will we end up calculating this value incorrectly if this happens?

Reply:
Its very much possible that certain resource groups are only available on certain coordinators and in that case we can end up getting empty result from the resource manager.